SrijanShovit/HealthLearning
# generic machine learning model comparison

### Description
This PR adds a generic machine learning model comparison script. The script allows for easy comparison of various machine learning models on both normalized and standardized data. It includes functionalities such as model training, evaluation, hyperparameter tuning, and visualization of performance metrics.

### Related Issues


### Changes
- Added a Python script for comparing machine learning models.
- Implemented support for both normalization and standardization of data.
- Included model training, evaluation, hyperparameter tuning functionalities.
- Visualized performance metrics including ROC curves and confusion matrices.

### Testing Instructions
1. Clone this branch.
2. Ensure Python is installed on your system.
3. Install the required dependencies using `pip install`.
4. Run the script `python model_comparison.py`.
5. Verify that the models are trained, evaluated, and visualized correctly.

### Screenshots (if applicable)
N/A

### Additional Context
This PR is part of my contribution to GSSOC.

### Checklist
